File: src/options_types/OfflineMode.zig
pub const OfflineMode = enum {
    online,
    latest,
    offline,
};

pub const Prefer = bun.ComptimeStringMap(OfflineMode, .{
    &.{ "offline", OfflineMode.offline },
    &.{ "latest", OfflineMode.latest },
    &.{ "online", OfflineMode.online },
});

const bun = @import("bun");

File: src/options_types/offline_mode.rs
#[repr(u8)]
#[derive(Copy, Clone, PartialEq, Eq, Debug, Default)]
pub enum OfflineMode {
    #[default]
    Online,
    Latest,
    Offline,
}

bun_core::comptime_string_map! {
    pub static PREFER: OfflineMode = {
        b"offline" => OfflineMode::Offline,
        b"latest" => OfflineMode::Latest,
        b"online" => OfflineMode::Online,
    };
}
